components/dashtable_v2.js

```diff
@@ -76,15 +78,17 @@ export default function GlobalDashboardTable(props) {
       activity: studentActivity,
       progress: percentageCompletion,
       detail: (
-        <a
-          // TODO:
-          href={
-            `/dashboard/v2/details/${props.classroomId}/` + `${studentName}`
-          }
+        <Link
+          href={{
+            pathname: '/dashboard/v2/details',
+            query: {
+              studentName: studentName,
+              classroomName: classroomName
+            } // the data
+          }}
         >
-          {' '}
-          details{' '}
-        </a>
+          details
+        </Link>
```

Comment:
Hi @Komal914, note that the file structure of the details directory 
`details/[studentName].js./index.js`, we can make use of Next.js' built-in[ dynamic routing](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes) and the use of [data fetching](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props) with the getServerSideProps() function => see updated code review for index.js file.

This is the beauty of Next.js (built-in handling for reserved characters in URL params). It takes care of the messiness (encoding/decoding) under the hood. When you have a square bracket file, whatever data/variable you want the sqaure bracket to absorb, will then be accessible in that file using something like:
`context.params.<whatever your square bracket file name is>`

Lines 81-91 should be changed to the following (remember the file structure details/[studentName].js/index.js):

`<Link
          href={{ pathname: `/dashboard/v2/details/${studentName}`, //  details/[id.js] ends up being details/[${studentName}].js
          query: {
            classroomName: classroomName
          }}}
        >
          details
        </Link>`
        

${studentName} gets fed into the square brackets => [id].js = [string stored in studentName].js, and magically you'll note, the reserved characters are preserved in the URL! Additionally, we can keep the query for classRoom name.
 
You can access the classroomName URL query param the same way as before, and you can access the studentName with context.params.id. (Note it's not context.params.studentName because the current file structure is as follows: details/[id].js/index.js. You can rename [id].js to be [studentName].js and then you would be able to use context.params.studentName).